rudder-agent/SOURCES/patches/fusioninventory/25395-windows-utf8.patch

```diff
@@ -0,0 +1,813 @@
+ 
+     if ($found) {
+-        my $lastupdate = hex2dec($found->{"Data/"}->{"/LastSuccessfulUpdate"});
++        my $lastupdate = hex2dec($found->{"Data"}->{"LastSuccessfulUpdate"});
+         if ($lastupdate && $lastupdate != 0xFFFFFFFF) {
+             my @date = localtime($lastupdate);
+             # Format BASE_VERSION as YYYYMMDD
+@@ -213,9 +213,9 @@
+                 "%04d%02d%02d",$date[5]+1900,$date[4]+1,$date[3]);
+         }
+         # Set expiration date only if we found a licence key type
+-        my $keytype = hex2dec($found->{"Data/"}->{"/LicKeyType"});
++        my $keytype = hex2dec($found->{"Data"}->{"LicKeyType"});
+         if ($keytype) {
+-            my $expiration = hex2dec($found->{"Data/"}->{"/LicDaysTillExpiration"});
++            my $expiration = hex2dec($found->{"Data"}->{"LicDaysTillExpiration"});
+             if (defined($expiration)) {
+                 my @date = localtime(time+86400*$expiration);
+                 $antivirus->{EXPIRATION} = sprintf(
+@@ -461,13 +461,10 @@
+ sub _getSoftwareRegistryKeys {
+     my ($base, $values, $callback) = @_;
+ 
+-    my $reg;
+     if (is64bit()) {
+-        $reg = getRegistryKey(
+-            path => 'HKEY_LOCAL_MACHINE/SOFTWARE/Wow6432Node/'.$base,
+-            wmiopts => { # Only used for remote WMI optimization
+-                values  => $values
+-            }
++        my $reg = getNewRegistryValues(
++            root   => "HKEY_LOCAL_MACHINE",
++            path   => 'SOFTWARE/Wow6432Node/'.$base
+         );
+         if ($reg) {
+             if ($callback) {
+@@ -478,12 +475,9 @@
+             }
+         }
+     }
+-
+-    $reg = getRegistryKey(
+-        path => 'HKEY_LOCAL_MACHINE/SOFTWARE/'.$base,
+-        wmiopts => { # Only used for remote WMI optimization
+-            values  => $values
+-        }
++    my $reg = getNewRegistryValues(
++        root   => "HKEY_LOCAL_MACHINE",
++        path   => 'SOFTWARE/'.$base
+     );
+     return ($callback && $reg) ? &{$callback}($reg) : $reg;
+ }
+diff --color -ruN fusion-source.orig/lib/FusionInventory/Agent/Task/Inventory/Win32/CPU.pm fusion-source/lib/FusionInventory/Agent/Task/Inventory/Win32/CPU.pm
+--- fusion-source.orig/lib/FusionInventory/Agent/Task/Inventory/Win32/CPU.pm	2020-08-04 09:36:42.000000000 +0200
++++ fusion-source/lib/FusionInventory/Agent/Task/Inventory/Win32/CPU.pm	2024-08-30 18:26:49.835436445 +0200
+@@ -53,12 +53,11 @@
+     my @dmidecodeInfos = $remotewmi || Win32::GetOSName() eq 'Win2003' ?
+         () : getCpusFromDmidecode();
+ 
+-    # the CPU description in WMI is false, we use the registry instead
+-    my $registryInfos = getRegistryKey(
+-        path   => "HKEY_LOCAL_MACHINE/Hardware/Description/System/CentralProcessor",
+-        wmiopts => { # Only used for remote WMI optimization
+-            values  => [ qw/Identifier ProcessorNameString VendorIdentifier/ ]
+-        }
++    my $registryInfos = getNewRegistryAll(
++        logger => $params{logger},
```

Comment:
Why is the logger only passed here ? What does it log ?
If the logging is automatic, I think that we should pass it whenever we use the getNewRegistryAll function in our patch